compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -27,31 +26,35 @@ impl MutVisitor for Marker {
 }
 
 /// An iterator over the token trees in a delimited token tree (`{ ... }`) or a sequence (`$(...)`).
-enum Frame {
-    Delimited { forest: Lrc<mbe::Delimited>, idx: usize, span: DelimSpan },
-    Sequence { forest: Lrc<mbe::SequenceRepetition>, idx: usize, sep: Option<Token> },
+enum Frame<'a> {
+    Delimited {
+        tts: &'a [mbe::TokenTree],
+        delim_token: token::DelimToken,
+        idx: usize,
+        span: DelimSpan,
+    },
+    Sequence {
+        tts: &'a [mbe::TokenTree],
+        idx: usize,
```

Comment:
This `tts+idx` pair looks like [std](https://doc.rust-lang.org/stable/std/index.html)::[slice](https://doc.rust-lang.org/stable/std/slice/index.html)::[Iter](https://doc.rust-lang.org/stable/std/slice/struct.Iter.html#), which already implements `Iterator` so the custom `Iterator` impl would no longer be needed if it's used.

Reply:
This is a good observation, but I hit a problem [here](https://github.com/rust-lang/rust/blob/f6cef572d6d75a3b7dea5f496d3bd981986e94ca/compiler/rustc_expand/src/mbe/transcribe.rs#L125-L135) when I tried to make the change. For sequences we might re-iterate over them, and I don't think there's a nice way to reset a `std::slice::Iter`. So I think the existing code will have to stay.